### DISCUSSION TITLE
Cloudflare and Wireguard

### DISCUSSION THREAD

**Discussion by joaopedros2** [homeassistant-apps/app-cloudflared#252] [Q&A]
I want to use wireguard to access to all network in my home. Before cloudflare I used Duckdns and worked fine.
I have a domain from cloudflare and I created a subdomain with my external IP address and without proxy. It works fine but the problem is when my external IP change, I have to change it in cloudflare account to.
It's possible to use your addon to update the public IP automatically on cloudflare account?

**Comment by AlecRust** [comment]
I think the Cloudflare Home Assistant integration is what you're after here: https://www.home-assistant.io/integrations/cloudflare/

**Reply by joaopedros2** [reply]
I try to use the integration with all configs recommended on api token, but I don't get IP updates on Cloudflare. 
I saw that the integration has shown some problems but I don't know if it's fit my case:

https://github.com/home-assistant/core/issues?q=cloudflare+label%3A%22integration%3A+cloudflare%22

**Reply by brenner-tobias** [reply]
You can use this add-on and wireguard, though I would love to challenge why you would want to do that.
With your domain in Cloudflare and the Cloudflare proxy + tunnel, you can archive everything (and more / better) without using Wireguard.
You can create you Cloudflare application in [Cloudflare Zero Trust](https://developers.cloudflare.com/cloudflare-one/) and define the access rules. So for example, you can define that people have to[ join WARP](https://developers.cloudflare.com/cloudflare-one/identity/devices/warp-client-checks/) (something like a VPN) with the WARP app or just define to use another layer of 2FA.
This is all free and works like a charm, so no need from my perspective for anything else :-)

**Comment by joaopedros2** [comment]
my first concern is to use the vpn connection to see Streaming and the
cloudflare tunnel has restrictions, right? but my question is if you use
cloudflare's DNS and wireguard the streaming problem is the same.

A quarta, 16/11/2022, 19:43, Tobias Brenner ***@***.***>
escreveu:

**Reply by brenner-tobias** [reply]
yes, so if you need your VPN to access something outside of your local network (e.g. stream via your home network from Netflix), then you should use a VPN software, agreed. Now to do that with Cloudflare, I suggest you use the [above mentioned integration](https://www.home-assistant.io/integrations/cloudflare/) to update your Cloudflare DNS-record with your home IP address.
